Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,170 @@
+     * Returns the queue name properly treated with queue prefix.
      *
-     * @return bool
+     * @param EndpointInterface $endpoint
+     * @return string
      */
-    public function consume(EndpointInterface $endpoint)
+    protected function getQueueName(EndpointInterface $endpoint): string
     {
-        $handler = new AmqpQueueHandler($endpoint, (int) $this->expirationCount, $this->format, $this->serializer);
-        $handler->setExceptionHandler($this->getExceptionHandler());
+        $options = $endpoint->getOptions();
 
-        if ($this->smartesbHelper) {
-            $handler->setSmartesbHelper($this->smartesbHelper);
-        }
-        if ($this->logger) {
-            $handler->setLogger($this->logger);
-        }
+        return "{$options[QueueProtocol::OPTION_PREFIX]}{$options[QueueProtocol::OPTION_QUEUE_NAME]}";
+    }
 
-        try {
-            $this->manager->getQueue($this->getQueueName($endpoint))->consume($handler);
-        } finally {
-            $this->manager->disconnect();
-        }
+    /**
+     * {@inheritdoc}
+     */
+    protected function cleanUp(EndpointInterface $endpoint)
+    {
+        $this->getQueueDriver($endpoint)->destroy($this->getName());
+    }
 
-        return true;
+    /**
+     * {@inheritdoc}
+     */
+    protected function confirmMessage(EndpointInterface $endpoint, MessageInterface $message)
+    {
+        /*
+         * Verify first that we have a QueueMessageInterface. If not, pass null and pray that the driver is
+         * keeping track of which messages need to be acked.
+         */
+        $this->getQueueDriver($endpoint)->ack($message instanceof QueueMessageInterface ? $message : null);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getName()
+    public function asyncConsume(EndpointInterface $endpoint, callable $callback)
     {
-        return 'async_queue_consumer';
+        $queueName = $this->getQueueName($endpoint);
+        $this->getQueueDriver($endpoint)->consume($this->getName(), $queueName, $callback);
     }
 
-    protected function getQueueName(EndpointInterface $endpoint): string
+    /**
+     * {@inheritdoc}
+     */
+    public function waitNoBlock(EndpointInterface $endpoint)
     {
-        $options = $endpoint->getOptions();
+        $this->getQueueDriver($endpoint)->waitNoBlock();
+    }
 
-        return "{$options[QueueProtocol::OPTION_PREFIX]}{$options[QueueProtocol::OPTION_QUEUE_NAME]}";
+    /**
+     * {@inheritdoc}
+     */
+    public function wait(EndpointInterface $endpoint)
+    {
+        $this->getQueueDriver($endpoint)->wait();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function process(EndpointInterface $queueEndpoint, MessageInterface $message)
+    {
+        // If we used a wrapper to queue the message, that the handler doesn't understand, unwrap it
+        if ($message instanceof MessageInterface && !($queueEndpoint->getHandler() instanceof QueueMessageHandlerInterface)) {
```

Comment:
In this case we should make sure the $message is an instance of `QueueMessageInterface`

Reply:
Done.